index.js

```diff
@@ -1,21 +1,15 @@
 var loaderUtils = require('loader-utils');
 var MessageFormat = require('messageformat');
 
-module.exports = function(content) {
-  var query = loaderUtils.parseQuery(this.query);
-  var locale = query.locale || 'en';
-  var messages = (Array.isArray(this.inputValue) && typeof this.inputValue[0] === 'object') ? this.inputValue[0] : this.exec(content, this.resource);
```

Comment:
Actually, even on Webpack 2+ they're required if you want to pipe the JSON file into any loader expecting JS. The only difference for Webpack 2+ is that the final loader can export JSON (or you can use no loaders) and Webpack itself will understand that. It's a little weird, but I just checked locally and that's what I found.

And yeah, I did notice that `this.exec` is deprecated, which is sad, because I'm not really sure what the alternative is.